File: k_guard_shop_fe/src/service/NewsService.js
import axios from "axios";
import * as ProductService from "../service/ProductService"
export const getAllNews = async () => {
    const res = await axios.get("http://localhost:8080/api/news")
    return res;
}
export const getNewsById = async (id) => {
    const res = await axios.get("http://localhost:8080/api/news/" + (+id))
    return res;
}

File: k_guard_shop_be/src/main/java/com/example/k_guard_shop_be/service/news/NewsService.java
package com.example.k_guard_shop_be.service.news;

import com.example.k_guard_shop_be.model.News;
import com.example.k_guard_shop_be.repository.INewsRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import javax.persistence.Entity;

@Service
public class NewsService implements INewsService{
    @Autowired
    private INewsRepository iNewsRepository;

    @Override
    public Page<News> getAllNews(Pageable pageable) {
        return iNewsRepository.findAll(pageable);
    }

    @Override
    public News getNewsById(Long id) {
        return iNewsRepository.findById(id).get();
    }
}
